vncdotool/client.py

```diff
@@ -394,6 +403,7 @@ def commitUpdate(self, rectangles):
             d = self.deferred
```

Comment:
Thanks for the explanation. I think [_capture()](https://github.com/sibson/vncdotool/blob/3cadfcbd35c0dd85a55ce2cde0045c112f1c4b71/vncdotool/client.py#L241-L244) and [commitUpdate()](https://github.com/sibson/vncdotool/blob/3cadfcbd35c0dd85a55ce2cde0045c112f1c4b71/vncdotool/client.py#L393-L397) are trying to achieve the same as what you are doing with ```screen_updated```.  I wonder if there is an order of ops or other unexpected call to ```commitUpdate()``` that is causing it to prematurely fire the callback. In particular a pseduo-cursor or empty rect update.

Reply:
Yeah, at some point I changed `_capture` to return `self` instead of `self.deferred` thinking that would stop the callback getting called prematurely, but it didn't, so I'm not sure.

When I added some extra debug logging, I didn't see extra calls to any of the methods, the callback was just getting called before `commitUpdate`.

I didn't spend a lot of time trying to figure it out what exactly the cause of the problem was once I figured out how to work around it. I may have time to take another look this weekend.